adapters/jms-adapter-generator/src/test/java/uk/gov/justice/raml/jms/core/JmsEndpointGeneratorTest.java

```diff
@@ -0,0 +1,407 @@
+    }
+
+    @Test
+    public void shouldCreateAnnotatedJmsEndpointWithMessageSelectorContainingOneCommandWithAPost() throws Exception {
+        generator.run(
+                raml()
+                        .with(resource()
+                                .with(action()
+                                        .with(ActionType.POST)
+                                        .withMediaType("application/vnd.structure.commands.test-cmd+json")))
+                        .build(),
+                configurationWithBasePackage(BASE_PACKAGE));
+
+        Class<?> clazz = compiler.compiledClassOf(BASE_PACKAGE);
+        assertThat(clazz.getAnnotation(MessageDriven.class), is(notNullValue()));
+        assertThat(clazz.getAnnotation(MessageDriven.class).activationConfig(),
+                hasItemInArray(allOf(propertyName(equalTo("messageSelector")),
+                        propertyValue(equalTo("CPPNAME in('structure.commands.test-cmd')")))));
+    }
+
+    @Test
+    public void shouldOnlyCreateMessageSelectorForPostActionAndIgnoreAllOtherActions() throws Exception {
+        generator.run(
+                raml()
+                        .with(resource()
+                                .with(action()
+                                        .with(ActionType.POST)
+                                        .withMediaType("application/vnd.structure.commands.test-cmd1+json"))
+                                .with(action()
+                                        .with(ActionType.GET)
+                                        .withMediaType("application/vnd.structure.commands.test-cmd2+json"))
+                                .with(action()
+                                        .with(ActionType.PUT)
+                                        .withMediaType("application/vnd.structure.commands.test-cmd3+json"))
+                                .with(action()
+                                        .with(ActionType.DELETE)
+                                        .withMediaType("application/vnd.structure.commands.test-cmd4+json"))
+                                .with(action()
+                                        .with(ActionType.HEAD)
+                                        .withMediaType("application/vnd.structure.commands.test-cmd5+json"))
+                                .with(action()
+                                        .with(ActionType.OPTIONS)
+                                        .withMediaType("application/vnd.structure.commands.test-cmd6+json"))
+                                .with(action()
+                                        .with(ActionType.PATCH)
+                                        .withMediaType("application/vnd.structure.commands.test-cmd7+json"))
+                                .with(action()
+                                        .with(ActionType.TRACE)
+                                        .withMediaType("application/vnd.structure.commands.test-cmd8+json")))
+                        .build(),
```

Comment:
There is a static convenience action() method that can take an action type and a mediatype in one go, which would make this shorter.